Azure-Samples/azure-search-openai-demo
# Default log level for dev vs prod

## Purpose

I've noticed I'm often telling people how to change the logging level. This PR makes a default level of INFO locally and WARNING in prod, since it's common to want different levels in those two environments. It can still be customized via env variables.

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* start.sh, see INFO level logs like:
* 
```
WARNING:azure.identity.aio._credentials.environment:Incomplete environment configuration. These variables are set: AZURE_TENANT_ID
INFO:azure.identity.aio._credentials.managed_identity:ManagedIdentityCredential will use IMDS
INFO:azure.core.pipeline.policies.http_logging_policy:Request URL: 'http://169.254.169.254/metadata/identity/oauth2/token?api-version=REDACTED&resource=REDACTED'
Request method: 'GET'
Request headers:
    'User-Agent': 'azsdk-python-identity/1.14.0 Python/3.11.5 (Linux-5.15.49-linuxkit-pr-aarch64-with-glibc2.36)'
No body was attached to the request
INFO:azure.identity.aio._credentials.chained:DefaultAzureCredential acquired a token from AzureDeveloperCliCredential
```

* deploy, only see WARNING level (none typically, if things are configured correctly)
